### DISCUSSION TITLE
SD to Headphones on ESP32 A1S AC101

### DISCUSSION THREAD

**Discussion by vacationvengarl** [pschatzmann/arduino-audiokit#35] [Q&A] [answered]
Hello, sorry if I'm doing something wildly wrong but I'm very new at this. I've checked that my model is AC101 and I have set it in the audiokitsetting.h file.
I'm trying to play .mp3 files from SD card to the headphone port using the player-sd-audiokit example. How do I modify the sketch to make this possible?
I'm also confused as to where the sound should be going out by default. I understand that the output is set to TX_MODE, but what pin is that on the board, TX0? Rout+? Linein? I have a speaker with +/- wires that I can also test with. 
Sorry again, I'm inexperienced at coding but would very much like to get this project working. Thank you

**Comment by pschatzmann** [comment]
The sketch should work w/o any changes. 

The output goes both to the speakers and aux. If you plug in the headphone, the speakers are automatically deactivated.
If something is not working you should find an error message in the log

**Reply by vacationvengarl** [reply]
Everything compiles w/o errors, but still no sound. 
When I press the reset button on the board, I get these errors in the log: 

rst:0x1 (POWERON_RESET),boot:0x1f (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0018,len:4
load:0x3fff001c,len:1216
ho 0 tail 12 room 4
load:0x40078000,len:10944
load:0x40080400,len:6388
entry 0x400806b4
[I] AudioTypes.h : 48 - sample_rate: 44100
[I] AudioTypes.h : 49 - channels: 2
[I] AudioTypes.h : 50 - bits_per_sample: 16
[I] I2SConfig.h : 95 - rx/tx mode: TX
[I] I2SConfig.h : 96 - port_no: 0
[I] I2SConfig.h : 97 - is_master: Master
[I] I2SConfig.h : 98 - sample rate: 44100
[I] I2SConfig.h : 99 - bits per sample: 16
[I] I2SConfig.h : 100 - number of channels: 2
[I] I2SConfig.h : 101 - i2s_format: I2S_STD_FORMAT
[I] I2SConfig.h : 103 - auto_clear:165
[I] I2SConfig.h : 110 - buffer_count:10
[I] I2SConfig.h : 111 - buffer_size:512
[I] I2SConfig.h : 115 - pin_bck: 14
[I] I2SConfig.h : 116 - pin_ws: 15
[I] I2SConfig.h : 117 - pin_data: -1515870811
[W] AudioKit.h : 46 - sd_active = true
Error:   reset failed!
Error:   codec init failed!
Error:   audio_hal_init
[E] AudioKit.h : 189 - begin faild: please verify your AUDIOKIT_BOARD setting: 6


I noticed this is the same error from frequently asked questions, but I've tried the other 2 board settings for the audiokit board already. For setting 5, it doesn't compile. And for setting 7, it compiles, but I get even more errors in the log. What should I do from here?

**Reply by pschatzmann** [reply]
When did you buy your board ?
What number is printed next to the V2.2 ?

I suggest that you start your tests with the streams-generator-audiokit.ino test. This way you can avoid the complexity around the SD drive...

**Reply by vacationvengarl** [reply]
Board was purchased about a week ago. Where would I find the number next to V2.2? It doesn't say V2.2 anywhere on the outside of the board. I'm also afraid of messing up the board by taking off the shield to look inside the ESP32-A1S unit.
This is the board I purchased: https://www.amazon.com/gp/product/B083FS43BL/ref=ask_ql_qh_dp_hza

**Comment by pschatzmann** [comment, marked as answer]
It is printed below ESP32 Audiokit
Is there really V2.1 2287 printed on it like on the picture ?

Never seen this board before and I don't have any information about this Taidacent board. It seems that you will need to figure out all the relevant pin assignments first and implement your own driver.

If you are lucky it is a LyraT clone, so try it with these settings first.

Here is an overview of the existing key pins across the supported boards: https://github.com/pschatzmann/arduino-audiokit/wiki/GPIO-Overview-by-Selected-Board

**Reply by vacationvengarl** [reply]
No sorry, the picture on amazon is not accurate, but I managed to find where it says the board version. Right under the 5 switches, it says ESP32 Audio Kit V2.2 A149

**Reply by pschatzmann** [reply]
That should work as documented with setting 5

**Reply by vacationvengarl** [reply]
This is what my board looks like. I'm trying the streams-generator-audiokit sketch right now with setting 5, hopefully it works.
![20220718_132615](https://user-images.githubusercontent.com/109392883/179612045-26808839-6747-420a-a9a0-45f9c83c742a.jpg)

**Reply by vacationvengarl** [reply]
Thank you so much for your help! The board works through stream-generator and player-SD now with setting 5. I don't know what went wrong the last time I tried compiling with setting 5, but everything works as intended now and I'm getting proper updates in the log.

**Comment by TangerineDreamer** [comment]
I have this one from aliexpress ,under the name "Jessinie,the antenna copper is blacked like the photo
and  i got it working with setting = 5.
